src/app/widget-builder/components/widgets/search-results-widget/search-results-widget-edit.component.ts

```diff
@@ -216,6 +221,34 @@ export class SearchResultsWidgetEditComponent extends BaseWidgetEditComponent {
     });
   }
 
+  // method which loads the active publishers from the widget settings and handles the checbox states
+  addPublishersControl() {
+    const activePublishers = this.widget.settings.detail_page.articles.publishers;
+    let arr = [];
+    if (this.publishers.length) {
+      arr = this.publishers.map( publisher => {
+        const active = activePublishers.includes(publisher);
+        return this.formBuilder.control(active);
+      });
+    }
+    return this.formBuilder.array(arr);
+  }
+
+  // method which updates the selected publishers
+  getSelectedPublishers() {
```

Comment:
If we would rename this method to `updateSelectedPublishers()`, we can easily remove the redundant comment on the line above